### PR TITLE
Add `hn-introspect` command

### LINKED ISSUE #25: An equivalent of the `hc-introspect` command

### PR DESCRIPTION
**Pull request by ThetaSinner**
Closes #25

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -109,17 +109,18 @@ jobs:
...
-        id: extract_branch
 
       - name: Init from template
         if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |
           cd $(mktemp -d)
           echo "Fetching template from branch ${{ steps.extract_branch.outputs.branch }}"
           nix flake init -t "github:holochain/holonix/${{ steps.extract_branch.outputs.branch }}#${{ matrix.template }}"
-          nix develop -c holochain --version
+          nix develop --override-input holonix "github:holochain/holonix/${{ steps.extract_branch.outputs.branch }}" -c holochain --version
```
Comment (ThetaSinner) [inline .github/workflows/build.yaml:122]: The test should be that the template works with any changes defined in Holonix on the current branch. On `main` the override is redundant but still valid.

File: flake.nix [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
   # specify all input dependencies needed to create the outputs of the flake
   inputs = {
-    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+    nixpkgs.url = "github:nixos/nixpkgs?ref=24.05";
```
Comment (ThetaSinner) [inline flake.nix:7]: Sneaking this in, I'd rather be pinned to some stable packages than use unstable, unless there was a reason to do so?

File: flake.nix [hunk 3]
```diff
@@ -254,6 +254,39 @@
...
+                echo "Lair keystore : $(lair-keystore --version)"
+              fi
+
+              if ! type "holochain" > /dev/null; then
+                echo "Holochain     : not installed"
+              else
+                echo "Holochain     : $(holochain --version)"
+
+                printf "\nHolochain build info: "
+                holochain --build-info | ${pkgs.jq}/bin/jq
```
Comment (ThetaSinner) [inline flake.nix:287]: Just to make the output easier to read, `jq` will format the JSON output and colour it.

File: templates/default/flake.nix [hunk 4]
```diff
@@ -21,6 +21,7 @@
           lair-keystore
           hc-launch
           hc-scaffold
+          hn-introspect
```
Comment (ThetaSinner) [inline templates/default/flake.nix:24]: Note that `hn-introspect` has to be included in the final shell, the same as other binaries. Putting it in the template is as close as we can get to 'we recommend including this please' :)

**Review by jost-s — APPROVED** [review]
This is all that `hn-introspect` did? Somewhat flattering name.
Cool, LGTM

**Comment by jost-s** [comment]
Wasn't there something with HDK and HDI versions?

**Comment by ThetaSinner** [comment]
Haha! It did do one thing I'd missed, which is to include Git revisions.

Old `hn-introspect` output:

```
holochain (holochain 0.4.0-dev.13): d7ef2a0947908ba1f859a535ef9ea31f89c2d355
lair-keystore (lair_keystore 0.4.5): 6a84ed490fc7074d107e38bbb4a8d707e9b8e066
hc-scaffold (holochain_scaffolding_cli 0.4000.0-dev.11): 6cb5a8d59d0eb844c4c727736a16d2f0c72caf0b
hc-launch (holochain_cli_launch 0.400.0-dev.0 (holochain 0.4.0-dev.12)): b788d346491f4749949ae1710d51508920b6ea07
```

New output:

```
hc-scaffold   : holochain_scaffolding_cli 0.4000.0-dev.10 (3241aaa)
hc-launch     : holochain_cli_launch 0.400.0-dev.0 (holochain 0.4.0-dev.12) (b788d34)
Lair keystore : lair_keystore 0.4.5 (6a84ed4)
Holochain     : holochain 0.4.0-dev.13 (d7ef2a0)

Holochain build info: {
  "git_info": null,
  "cargo_pkg_version": "0.4.0-dev.13",
  "hdk_version_req": "0.4.0-dev.11",
  "hdi_version_req": "0.5.0-dev.10",
  "lair_keystore_version_req": "0.4.5",
  "timestamp": "2024-07-19T16:11:55.211733946Z",
  "hostname": "localhost",
  "host": "x86_64-unknown-linux-gnu",
  "target": "x86_64-unknown-linux-gnu",
  "rustc_version": "rustc 1.78.0 (9b00956e5 2024-04-29)",
  "rustflags": "",
  "profile": "release"
}
```

So the HDK and HDI versions are included in the `--build-info` for Holochain. Which is great if you knew to run that, but I think it's better to just run it with the introspect. It's not trivially to parse anyway, so why not add a bit more information by default.

**Pull request merged**